### ISSUE TITLE
Please support ROB_200-007-0 (new device)

### ISSUE THREAD

**Issue by JHurk** [stas-demydiuk/domoticz-zigbee2mqtt-plugin#528]
ROB_200-007-0

https://www.zigbee2mqtt.io/devices/ROB_200-007-0.html

MQTT message with device information
info  2020-12-23 10:13:09: MQTT publish: topic 'zigbee2mqtt/0xbc33acfffe63351d', payload '{"battery":50,"device":{"applicationVersion":0,"dateCode":"NULL","friendlyName":"0xbc33acfffe63351d","hardwareVersion":0,"ieeeAddr":"0xbc33acfffe63351d","manufacturerID":4644,"manufacturerName":"Sunricher","model":"ROB_200-007-0","networkAddress":38905,"powerSource":"Battery","softwareBuildID":"2.5.3_r20","stackVersion":0,"type":"EndDevice","zclVersion":3}}'

**MQTT state change commands**
Triggered action (e.g. a button click). Value can be found in the published state on the action property. It’s not possible to read (/get) or write (/set) this value. The possible values are: on_1, off_1, brightness_move_up_1, brightness_move_down_1, brightness_stop_1, on_2, off_2, brightness_move_up_2, brightness_move_down_2, brightness_stop_2, on_3, off_3, brightness_move_up_3, brightness_move_down_3, brightness_stop_3, on_4, off_4, brightness_move_up_4, brightness_move_down_4, brightness_stop_4, on_5, off_5, brightness_move_up_5, brightness_move_down_5, brightness_stop_5, on_6, off_6, brightness_move_up_6, brightness_move_down_6, brightness_stop_6, on_7, off_7, brightness_move_up_7, brightness_move_down_7, brightness_stop_7, on_8, off_8, brightness_move_up_8, brightness_move_down_8, brightness_stop_8.

**Extra information:**
Zigbee2MQTT version 1.16.2 (commit #04c15f7)
zigbee-herdsman (0.13.37)
New_api enabled
Zigbee2MQTT domoticz plugin downloaded latest master (so not the latest release)
I have read about the Universal Adapter and thought with all the above it would work, but the device is still not showing up in the Device list. In the log it stated that the device was not supported by the plugin and I should make a request (this issue). I updated the plugin to the latest master, with no luck.

**Comment by JHurk** [comment]
Got it to work with the Universal Adapter. I forgot to remove the plugin from Domoticz/plugins and reinstall the plugin from scratch (all settings are still there).
I only had a few 'devices' when the Universal Adapter was activated. These are devices I already have in Domoticz but are now double (_state is new, _action was the old device). 5 minutes and this was solved/renamed and adjusted in Domoticz.

Now I need to figure out how to set the commands for each of the 8 buttons.
Think this will help:
https://www.domoticz.com/wiki/Domoticz_API/JSON_URL%27s#Selector_Switch

**Comment by LPCevaal** [comment]
Hello, I have the same device. I have been able to get it to show up as a single selector switch in the plugin, but the state is always ON, no matter which button I press. I can see that every click is registered as "last seen" is updated as soon as a press a button.

Have you been able to get the data from the different button clicks?

**Comment by JHurk** [comment]
Will try and explain it in a step by step way so you understand what happens and needs to be done. THis way you can experiment and adjust it any way you like.

When in the switch tab of Domoticz you click 'aanpassen' (adjust):
<img width="410" alt="Schermafbeelding 2020-12-29 om 09 33 00" src="https://user-images.githubusercontent.com/7835370/103272305-d0299f00-49bc-11eb-8038-5a06e2644baf.png">

You will see the details of the 8 button selector, with the first table 'Keuzeniveau' being all the commands the physical buttons can sent to Domoticz. For example:
Niveau 10 = On1 - which is the top left button of the device
Niveau 20 = Off1 - which is the top right button of the device
etc.
<img width="340" alt="Schermafbeelding 2020-12-29 om 09 47 38" src="https://user-images.githubusercontent.com/7835370/103272416-18e15800-49bd-11eb-9b6f-2ce37f550f37.png">

So now you need to assign each command which the device sends to Domoticz, to an action. For example 'turn on a light' 
This is done on the second, buttom table 'Selectie acties' (Selector actions). In this table you see the 'niveau' and have to assign a command to each niveau (command). 
<img width="1162" alt="Schermafbeelding 2020-12-29 om 09 53 10" src="https://user-images.githubusercontent.com/7835370/103272575-87261a80-49bd-11eb-9ce4-f62a5597cda3.png">

Niveau 10 in my case will give the following command: 
http://192.168.1.7:8084/json.htm?type=command&param=switchscene&idx=1&switchcmd=On
The things to notice here are:
- 192.168.1.7 = the ip of the machine where Domoticz runs
- 8084 = the port of Domoticz (standard is 8080 if I recollect correct, you can find this in you settings)
- param=switchscene = switchscene is the parameter which needs to be given when you want to switch a scene or group instead of a single light or device
- idx=1 = the idx of the scene/group I want to control (in my case IDX 1 is 'outside lighting')
- switchcmd=On = the command I want to give to the scene/group (or device), On in this case.
So basically you can assign any command to each button in this way, works really easy.
A complete guide to all commands and options can be found here:
https://www.domoticz.com/wiki/Domoticz_API/JSON_URL%27s
It is a long read but once you understand the bigger picture it will only be needed as a reference guide.

A simple URL to turn a light on will look like this;
http://<ip or domain>:<port>/json.htm?type=command&param=switchlight&idx=<IDX of the device>&switchcmd=On
To turn it of the URL looks like this:
http://<ip or domain>:<port>/json.htm?type=command&param=switchlight&idx=<IDX of the device>&switchcmd=Off

Please let me know of this helped and if not, where you are stuck so I might be able to help you in that particular step.

**Comment by LPCevaal** [comment]
Thank you very much for the excellent and extensive explanation! I will give it a try, but I am sure that with this explanation it should be a breeze. Good to be able to put all those new buttons to a good use :-)
